stdlib/public/core/Integers.swift.gyb

```diff
@@ -2388,21 +2416,20 @@ ${unsafeOperationComment(x.operator)}
   /// - Parameter source: An integer to convert to this type.
   @inline(__always)
   public init<T : BinaryInteger>(truncatingIfNeeded source: T) {
-    if Self.bitWidth <= ${word_bits} {
-      self = Self.init(_truncatingBits: source._lowWord)
+    if Self.bitWidth <= ${word_bits} || source.bitWidth <= ${word_bits} {
```

Comment:
I think the 32-bit failures are caused by this second clause -- `init(_truncatingBits:)` uses zero extension, so conversions to Int64 may fail in weird ways on those platforms.

Reply:
Yeah... Looks like it.